crates/router_derive/src/macros/to_encryptable.rs

```diff
@@ -0,0 +1,290 @@
+
+                if is_option || *self == Self::Updated {
+                    quote! { self.#field_ident.map(|s| map.insert(#field_ident_string.to_string(), s)) }
+                } else {
+                    quote! { map.insert(#field_ident_string.to_string(), self.#field_ident) }
+                }
+            })
+        });
+
+        let from_encryptable_impl = inner.iter().flat_map(|field| {
+            get_field_type(field.ty.clone()).map(|field_ty| {
+                let is_option = field_ty.eq("Option");
+                let field_ident = &field.ident;
+                let field_ident_string = field_ident.as_ref().map(|s| s.to_string());
+
+                if is_option || *self == Self::Updated {
+                    quote! { #field_ident: map.remove(#field_ident_string) }
+                } else {
+                    quote! {
+                        #field_ident: map.remove(#field_ident_string).ok_or(
+                            error_stack::report!(common_utils::errors::ParsingError::EncodeError(
+                                "Unable to convert from HashMap",
+                            ))
+                        )?
+                    }
+                }
+            })
+        });
+
+        quote! {
+            impl ToEncryptable<#gen1, #gen2, #gen3> for #impl_st {
+                fn to_encryptable(self) -> FxHashMap<String, #gen3> {
+                    let mut map = FxHashMap::with_capacity_and_hasher(#map_length, Default::default());
+                    #(#to_encryptable_impl;)*
+                    map
+                }
+
+                fn from_encryptable(
+                    mut map: FxHashMap<String, Encryptable<#gen2>>,
+                ) -> CustomResult<#gen1, common_utils::errors::ParsingError> {
+                    Ok(#gen1 {
+                        #(#from_encryptable_impl,)*
+                    })
+                }
+            }
+        }
+    }
+}
+
+/// This function generates the temporary struct and ToEncryptable impls for the temporary structs
+fn generate_to_encryptable(
+    struct_name: Ident,
+    fields: Vec<Field>,
+) -> syn::Result<proc_macro2::TokenStream> {
+    let struct_types = [
+        ("Decrypted", StructType::Decrypted),
+        ("DecryptedUpdate", StructType::DecryptedUpdate),
+        ("FromRequestEncryptable", StructType::FromRequest),
+        ("Encrypted", StructType::Encrypted),
+        ("UpdateEncryptable", StructType::Updated),
+    ];
+
+    let inner_types = get_field_and_inner_types(&fields);
+
+    let inner_type = inner_types.first().map(|(_, ty)| ty).ok_or_else(|| {
+        syn::Error::new(
+            proc_macro2::Span::call_site(),
+            "Please use the macro with attribute #[encrypt] on the fields you want to encrypt",
+        )
+    })?;
+
+    let structs = struct_types.iter().map(|(suffix, struct_type)| {
+        let name = format_ident!("{}{}", suffix, struct_name);
```

Comment:
Is the `suffix` supposed to be attached after the `struct_name` or before? If it's before, should we consider calling it `prefix`?